File: solutions/cpp/0049.cpp
class Solution {
 public:
  vector<vector<string>> groupAnagrams(vector<string>& strs) {
    vector<vector<string>> ans;
    unordered_map<string, vector<string>> map;

    for (const string& str : strs) {
      string s = str;
      sort(begin(s), end(s));
      map[s].push_back(str);
    }

    for (const auto& [_, value] : map)
      ans.push_back(value);

    return ans;
  }
};

File: solutions/java/0049.java
class Solution {
  public List<List<String>> groupAnagrams(String[] strs) {
    List<List<String>> ans = new ArrayList<>();
    Map<String, List<String>> map = new HashMap<>();

    for (final String str : strs) {
      char[] charArray = str.toCharArray();
      Arrays.sort(charArray);
      String s = String.valueOf(charArray);
      map.computeIfAbsent(s, k -> new ArrayList<>()).add(str);
    }

    return new ArrayList<>(map.values());
  }
}